server/backends/invocationdb/invocationdb.go

```diff
@@ -108,11 +109,19 @@ func (d *InvocationDB) CreateInvocation(ctx context.Context, ti *tables.Invocati
 // id and attempt number. It returns whether a row was updated.
 func (d *InvocationDB) UpdateInvocation(ctx context.Context, ti *tables.Invocation) (bool, error) {
 	updated := false
-	err := d.h.TransactionWithOptions(ctx, db.Opts().WithQueryName("update_invocation"), func(tx *db.DB) error {
-		result := tx.Where("`invocation_id` = ? AND `attempt` = ?", ti.InvocationID, ti.Attempt).Updates(ti)
-		updated = result.RowsAffected > 0
-		return result.Error
-	})
+	var err error
+	for retries, max_retries := 0, 5; retries < 5; retries++ {
```

Comment:
maybe using util/retry here would be better?

Reply:
oh, that's really nice.

Reply:
Done!